config/config.go

```
@@ -0,0 +1,78 @@
+	ChainId        uint8
+	Endpoint       string
+	Http           bool
+	Bridge         string
+	Erc20Handler   string
+	Erc721Handler  string
+	GenericHandler string
+	MaxGasPrice    *big.Int
+	GasMultiplier  *big.Float
+	GasLimit       *big.Int
+}
+
+type RawEVMConfig struct {
+	ChainId        uint8   `mapstructure:"id"`
+	Endpoint       string  `mapstructure:"endpoint"`
+	Http           bool    `mapstructure:"http"`
+	Bridge         string  `mapstructure:"bridge"`
+	Erc20Handler   string  `mapstructure:"erc20Handler"`
+	Erc721Handler  string  `mapstructure:"erc721Handler"`
+	GenericHandler string  `mapstructure:"genericHandler"`
+	MaxGasPrice    int64   `mapstructure:"maxGasPrice"`
+	GasMultiplier  float64 `mapstructure:"gasMultiplier"`
+	GasLimit       int64   `mapstructure:"gasLimit"`
+}
+
+func GetConfig(path string) (*EVMConfig, error) {
+	var rawConfig RawEVMConfig
+
+	viper.AddConfigPath(path)
+	viper.SetConfigName("config")
+	viper.SetConfigType("json")
+
+	err := viper.ReadInConfig()
+	if err != nil {
+		log.Warn().Msgf("Failed to read in the config file: %s", err)
+	}
+
+	err = viper.Unmarshal(&rawConfig)
+	if err != nil {
+		log.Warn().Msgf("Failed to unmarshal config into struct: %s", err)
+	}
+
+	log.Info().Msgf("%d", rawConfig.GasLimit)
+
+	config, err := parseConfig(&rawConfig)
+
+	return config, err
+}
+
+func parseConfig(rawConfig *RawEVMConfig) (*EVMConfig, error) {
+
+	config := &EVMConfig{
+		ChainId:        rawConfig.ChainId,
+		Endpoint:       rawConfig.Endpoint,
+		Http:           rawConfig.Http,
+		Bridge:         rawConfig.Bridge,
+		Erc20Handler:   rawConfig.Erc20Handler,
+		Erc721Handler:  rawConfig.Erc721Handler,
+		GenericHandler: rawConfig.GenericHandler,
+		GasLimit:       big.NewInt(rawConfig.GasLimit),
+		MaxGasPrice:    big.NewInt(int64(rawConfig.MaxGasPrice)),
+		GasMultiplier:  big.NewFloat(rawConfig.GasMultiplier),
+	}
+
+	log.Info().Msgf("gas limit after parse: %d", rawConfig.GasLimit)
+
+	return config, nil
```

Why does parseConfig need to return an error type?

Additionally, if you only need to return the pointer to the newly initialised struct, is it necessary to declare a new variable (config)?

```go
return &EVMConfig{
		ChainId:        rawConfig.ChainId,
		Endpoint:       rawConfig.Endpoint,
		Http:           rawConfig.Http,
		Bridge:         rawConfig.Bridge,
		Erc20Handler:   rawConfig.Erc20Handler,
		Erc721Handler:  rawConfig.Erc721Handler,
		GenericHandler: rawConfig.GenericHandler,
		GasLimit:       big.NewInt(rawConfig.GasLimit),
		MaxGasPrice:    big.NewInt(int64(rawConfig.MaxGasPrice)),
		GasMultiplier:  big.NewFloat(rawConfig.GasMultiplier),
	}
```

---

Also do not forget to add GasMultiplier functionality it could be found on Chainbridge repo

---

@vezenovm 

[config](https://github.com/ChainSafe/ChainBridge/blob/main/chains/ethereum/config.go#L54)

[functionality](https://github.com/ChainSafe/ChainBridge/blob/main/chains/ethereum/config.go#L131-L140)

---

I saw that as well. I thought it might be unnecessary as viper is unmarshaling into a float64 which can be then converted into a big float. While previously it was parsing from config fields purely from strings and transforming. I could change to load strings for each config field and parsing each directly into their respective type, but I went with this way for now. I also added in code where I set default gas values for the config, and also added using the `GasMultiplier` for when estimating gas in `client.go`.